frontend/src/store.ts

```diff
@@ -3166,7 +3166,51 @@ export function createStore(web3: Web3) {
           dispatch('fetchCharacterCosmetic', id)
         ]);
       },
+      async fetchTotalSmokeBombsOwned({ state }) {
+        const { SmokeBombConsumables } = state.contracts();
+        if(!SmokeBombConsumables || !state.defaultAccount) return;
+        return await SmokeBombConsumables.methods.getItemCount().call(defaultCallOptions(state));
+      },
+      async purchaseSmokeBomb50({ state, dispatch }) {
+        const { CryptoBlades, SkillToken, SmokeBombConsumables, Blacksmith } = state.contracts();
+        if(!CryptoBlades || !SmokeBombConsumables || !Blacksmith || !state.defaultAccount) return;
+
+        try {
+          await SkillToken.methods
+            .approve(CryptoBlades.options.address,  web3.utils.toWei('0.1', 'ether'))
```

Comment:
Minor suggestion, can't this price be passed by component using this method? Or we can use getSmokeBombPrice method, as it won't work if the price changes

Reply:
Please use `approveFee` function to approve skill spend - this handles reading the price from chain.

Reply:
used approveFeeFromAnyContract but had to do a modification which allows sending payToAddress as a param. Fee contract in this case (blacksmith) isn't the pay to address (cryptoblades).
Tested minting and we should be fine